### PR TITLE
Add RegisterMenuScreensEvent to register custom menu screens

### PR DESCRIPTION
**Pull request by ghost**
This adds a simple event making it easier to register screens.

This also renames SpawnPlacementsRegisterEvent to RegisterSpawnPlacementsEvent due to the 'conventions' forge made on the client refactors

### CODE REVIEW

File: src/main/java/net/minecraftforge/client/event/RegisterMenuScreensEvent.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    @ApiStatus.Internal
+    public RegisterMenuScreensEvent(Map<MenuType<?>, MenuScreens.ScreenConstructor<?, ?>> map)
+    {
+        this.map = map;
+    }
+
+    public void register(MenuType<?> menuType, MenuScreens.ScreenConstructor<?, ?> screenConstructor) {
+        map.put(menuType, screenConstructor);
+    }
+}
```
Comment (ChampionAsh5357) [inline src/main/java/net/minecraftforge/client/event/RegisterMenuScreensEvent.java:35]: LF at EOF.

File: src/main/java/net/minecraftforge/client/event/RegisterMenuScreensEvent.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{
+    private final Map<MenuType<?>, MenuScreens.ScreenConstructor<?, ?>> map;
+
+    @ApiStatus.Internal
+    public RegisterMenuScreensEvent(Map<MenuType<?>, MenuScreens.ScreenConstructor<?, ?>> map)
+    {
+        this.map = map;
+    }
+
+    public void register(MenuType<?> menuType, MenuScreens.ScreenConstructor<?, ?> screenConstructor) {
```
Comment (ChampionAsh5357) [inline src/main/java/net/minecraftforge/client/event/RegisterMenuScreensEvent.java:32]: Bracket on new line, missing documentation.

File: src/main/java/net/minecraftforge/common/property/Properties.java [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import net.minecraft.world.level.block.state.properties.BooleanProperty;
 import net.minecraftforge.client.model.data.ModelProperty;
 
+@Deprecated(since = "1.19.2", forRemoval = true)
```
Comment (ChampionAsh5357) [inline src/main/java/net/minecraftforge/common/property/Properties.java:12]: Why? This has nothing to do with the PR

File: src/main/java/net/minecraftforge/energy/EnergyStorage.java [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 /**
  * Reference implementation of {@link IEnergyStorage}. Use/extend this or implement your own.
- *
+ * <p>
```
Comment (ChampionAsh5357) [inline src/main/java/net/minecraftforge/energy/EnergyStorage.java:14]: Why?

File: src/main/java/net/minecraftforge/event/entity/RegisterSpawnPlacementsEvent.java [hunk 5]
```diff
@@ -41,12 +41,12 @@
  *
  *  Fired on the Mod bus {@link IModBusEvent}.<br>
  */
-public class SpawnPlacementRegisterEvent extends Event implements IModBusEvent
+public class RegisterSpawnPlacementsEvent extends Event implements IModBusEvent
```
Comment (ChampionAsh5357) [inline src/main/java/net/minecraftforge/event/entity/RegisterSpawnPlacementsEvent.java:44]: Breaking change, even bigger why

File: src/main/java/net/minecraftforge/items/SidedInventoryItemHandler.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        return insert.test(slot, stack) && handler.isItemValid(slot, stack);
+    }
+
+    @Override
+    public void setStackInSlot(int slot, @NotNull ItemStack stack)
+    {
+        handler.setStackInSlot(slot, stack);
+    }
+}
```
Comment (ChampionAsh5357) [inline src/main/java/net/minecraftforge/items/SidedInventoryItemHandler.java:52]: Not related to pr...again.

File: src/test/java/net/minecraftforge/debug/entity/SpawnPlacementTest.java [hunk 7]
```diff
@@ -31,13 +31,13 @@ public SpawnPlacementTest()
         }
     }
 
-    private void onSpawnPlacementRegister(SpawnPlacementRegisterEvent event)
+    private void onSpawnPlacementRegister(RegisterSpawnPlacementsEvent event)
```
Comment (ChampionAsh5357) [inline src/test/java/net/minecraftforge/debug/entity/SpawnPlacementTest.java:34]: Again, why

File: src/main/java/net/minecraftforge/client/event/RegisterMenuScreensEvent.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package net.minecraftforge.client.event;
```
Comment (ChampionAsh5357) [inline src/main/java/net/minecraftforge/client/event/RegisterMenuScreensEvent.java:1]: Missing license header.

**Review by ChampionAsh5357 — CHANGES_REQUESTED** [review]
This event currently does nothing. So how is it supposed to work?

Also, you need to justify your reasoning for adding this event. Why is this better than calling the method directly in an enqueued runnable within the client event?

**Comment by ghost** [comment]
will redo this PR

**Comment by FinnT730** [comment]
Why close it?
you can push changed to this PR

**Pull request closed without merging**